src/main/scala/mesosphere/mesos/TaskBuilder.scala

```diff
@@ -22,7 +21,8 @@ class TaskBuilder(
     runSpec: RunSpec,
     newTaskId: PathId => Task.Id,
     config: MarathonConf,
-    appTaskProc: Option[RunSpecTaskProcessor] = None) {
+    appTaskProc: Option[RunSpecTaskProcessor] = None,
+    rejectionCollectorOpt: Option[RejectOfferCollector]) {
```

Comment:
Instead of further parameterizing `TaskBuilder` I'd rather see "rejections" be emitted as the output from a build attempt.